### PR TITLE
tcbinfo: Update tcbinfo struct, Fix the compile warning

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
## Summary

Update tcbinfo struct, Fix the compile warning. Let's ignore the false nxstyle alarm:
```
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:387:4: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:394:9: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:399:18: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:404:9: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:409:9: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:414:9: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:429:4: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:493:4: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:530:4: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:567:4: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:603:4: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/tools/jlink-nuttx.c:642:4: error: Mixed case identifier found
Error: Process completed with exit code 1.
```

## Impact

No

## Testing

Testing on m55 with jtag enable.

### CODE REVIEW

File: arch/arm/src/arm/arm_tcbinfo.c [hunk 1]
```diff
@@ -63,6 +54,22 @@ const struct tcbinfo_s g_tcbinfo =
...
+
+const struct tcbinfo_s g_tcbinfo =
+{
+  TCB_PID_OFF,
+  TCB_STATE_OFF,
+  TCB_PRI_OFF,
+  TCB_NAME_OFF,
+  XCPTCONTEXT_REGS,
+  {
+    .p = g_reg_offs,
```
Comment (pkarashchenko) [inline arch/arm/src/arm/arm_tcbinfo.c:69]: uncompliant with C89, but since this is arch specific, I'm not sure if we need to care about it

Comment (xiaoxiang781216) [reply arch/arm/src/arm/arm_tcbinfo.c:69]: arm arch already has many similar construct, so we relax the check until we do the final decision?

Comment (xiaoxiang781216) [reply arch/arm/src/arm/arm_tcbinfo.c:69]: @pkarashchenko do you think it's reasonable to ignore this in the current state?

Comment (pkarashchenko) [reply arch/arm/src/arm/arm_tcbinfo.c:69]: Yes. Of course

**Pull request merged**